python/ray/serve/_private/deployment_state.py

```diff
@@ -1392,6 +1394,11 @@ def _failed_to_start_threshold(self) -> int:
             self._target_state.target_num_replicas * 3,
         )
 
+    @property
+    def is_available(self) -> bool:
+        """Whether the deployment is available for handles to send requests to."""
+        return self._curr_status_info.status != DeploymentStatus.DEPLOY_FAILED
```

Comment:
nit: this name is confusing. I would naiively think it'll be `False` if the deployment is still starting up

would suggest naming it something like `is_failed` or `is_unavailable`